### PR TITLE
feat: specify namespace on k8sgpt crd

### PR DESCRIPTION
**Pull request by singhiqbal1007**
Closes https://github.com/k8sgpt-ai/k8sgpt-operator/issues/270 

## 📑 Description

* added namespace selector on k8sgpt
* added namespace argument on grpc request

## ✅ Checks

- [x] My pull request adheres to the code style of this project
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [x] All the tests have passed

## ℹ Additional Information

Tested these changes on my local machine.
targetNamespace field is optional. If given then it looks for issues only in the given namespaces otherwise It looks for issues in all the namespaces.

### CODE REVIEW

**Comment by JuHyung-Son** [comment]
thanks for pr.
helm charts crd definition is not generating automatically. so you should manually add field in `chart/operator/templates/k8sgpt-crd.yaml`

File: pkg/client/analysis.go [hunk 1]
```diff
@@ -19,6 +19,7 @@ func (c *Client) ProcessAnalysis(deployment v1.Deployment, config *v1alpha1.K8sG
 		Explain:   config.Spec.AI.Enabled,
 		Nocache:   config.Spec.NoCache,
 		Backend:   config.Spec.AI.Backend,
+		Namespace: config.Spec.TargetNamespace,
```
Comment (arbreezy) [inline pkg/client/analysis.go:22]: @singhiqbal1007 shouldn't we set it only when it's configured in the K8sGPT CR, otherwise it will be an empty string ?

**Review by arbreezy — CHANGES_REQUESTED** [review]

File: pkg/client/analysis.go (hunk 1 above)
Comment (iqbalcoupa) [reply pkg/client/analysis.go:22]: @arbreezy When you don't pass anything it should get converted to an empty string. I couldn't find the code for the API object as API is stored in gRPC but you can see the screenshot below from vscode.
Screenshot 2024-01-24 at 9 02 25 AM

Additionally, I checked in the k8sgpt repository, we are using client-go to fetch resources so an empty string will fetch for all namespaces. Below is one example for pod.
https://github.com/k8sgpt-ai/k8sgpt/blob/main/pkg/analyzer/pod.go#L36

Also, I have tested the code locally and it's working as expected.

Comment (singhiqbal1007) [reply pkg/client/analysis.go:22]: @arbreezy When you don't pass anything it should get converted to an empty string. I couldn't find the code for the API object as API is stored in gRPC but you can see the screenshot below from vscode.
Screenshot 2024-01-24 at 9 02 25 AM

Additionally, I checked in the k8sgpt repository, we are using client-go to fetch resources so an empty string will fetch for all namespaces. Below is one example for pod.
https://github.com/k8sgpt-ai/k8sgpt/blob/main/pkg/analyzer/pod.go#L36

Also, I have tested the code locally and it's working as expected.

Comment (arbreezy) [reply pkg/client/analysis.go:22]: Cool, after resolving some conflicts we can merge, thanks for the PR !

**Comment by singhiqbal1007** [comment]
@arbreezy Resolved the conflicts.

**Review by arbreezy — APPROVED** [review]

**Pull request merged**